### ISSUE TITLE
JPA Transaction failures in long running (> 1 day) CAS server

### ISSUE THREAD

**Issue by frett** [Jasig/cas#1378]
I'm seeing this exception when trying to save a newly created TGT in the `JpaTicketRegistry` on 4.1.3:
```
org.hibernate.TransactionException: Already have an associated managed connection
```
I can get a full stack trace if necessary.

Based on preliminary googling it appears to be based on the fact that we have switched the JPATicketRegistry to use `@PersistenceContext(type = PersistenceContextType.EXTENDED)`. see: http://stackoverflow.com/questions/20756804/already-have-an-associated-managed-connection-error

I'm going to go back and investigate the JPA issue that this patch was added for to see if I can work out a better solution.

**Comment by mmoayyed** [comment]
If you have configured your transaction annotations correctly in xml, then you should be able to take that extended annotation from that field and see what happens.

**Comment by frett** [comment]
I was able to reproduce this issue on a fresh startup by logging in a couple times, going to the database and killing all active connections, then trying to login a few more times.

Sample audit message of the failure:

```
2015-12-26 11:05:35,512 INFO [org.jasig.inspektr.audit.support.Slf4jLoggingAuditTrailManager] - <Audit trail record BEGIN
=============================================================
WHO: casuser+password
WHAT: org.hibernate.TransactionException: Already have an associated managed connection
ACTION: TICKET_GRANTING_TICKET_NOT_CREATED
APPLICATION: CAS
WHEN: Sat Dec 26 11:05:35 EST 2015
CLIENT IP ADDRESS: 0:0:0:0:0:0:0:1
SERVER IP ADDRESS: 0:0:0:0:0:0:0:1
=============================================================
```

I was also able to fix the original issue in 4.1.2 by adding a few more transaction pointcuts in the sample overlay provided here: https://groups.google.com/a/apereo.org/d/msg/cas-user/-XM9OA2GKX8/Gso4sT53BgAJ

I did a brute force addition of all the pointcuts in 4.2, so i'm going to go filter through them to make sure everything is correct and will update the necessary documentation for 4.1.x once I figure out what the final result should be. I will also follow up with the original thread on the mailing list with the updated pointcuts.

a PR removing the offending `PersistenceContextType` will be incoming in a bit.
